### ISSUE TITLE
Remmina main repository integration

### ISSUE THREAD

**Issue by antenore** [alexmurray/remmina-search-provider#18]
Hi,

We're slowly integrating Remmina in the Gnome DE.

We'd like to add your search provider in the Remmina main code.

Do you mind helping us? We'd really love having a PR from you!!!

Thank you in advance!

**Comment by alexmurray** [comment]
Sure - the code functions as an extension to gnome-shell and is in Javascript at the moment - would you expect it to still work that way, OR would you have the functionality reimplemented in Remmina directly (ie. have a dbus search provider implemented in Remmina's gnome-plugins?) - I think this will perform a bit slowly since in that case dbus will have to launch remmina each time to do the search - but if it stays as a separate js extension then it is already running.

**Comment by antenore** [comment]
Yes, I prefer the js extension as well. Where it should be placed on the file system and how to activate it? Sorry I didn't RTFM

**Comment by mygithubthrowaway** [comment]
So, this extension is in remmina main code?

**Comment by alexmurray** [comment]
Not that I know of... @antenore are you still interested in having this upstream? Apologies this slipped off my radar.

**Comment by antenore** [comment]
Not yet and I'm not sure how to implement it. It's a matter of time, to understand before how it works and do the needed.

**Comment by antenore** [comment]
To better explain the situation.
The search provider should be used at a user level and not system wide, so that the user can turn it on or off based on his/her needs.
Doing this involves:

1. Adding a new option in Remmina
2. Adding the code that install and activate the search provider on demand.

Every time I start coding something I get lost in reading the Gnome spech (well the code and code documentation mainly), to understand

1. Where should I place the search provider (which directories)?
2. How to activate or deactivate it?

As far as I know a search provider can be set on/off directly with Gnome, so probably I don't even need the option in Remmina, but still, it's not clear for me how can I activate it from my code, so that at least it would be activated by default.

The thing is, if I install the search provider and the average user doesn't know it exist, it won't probably never use it, so having an option will make things more "visible".

If you have any thoughts...

**Comment by antenore** [comment]
https://developer.gnome.org/SearchProvider/

Found something :-)
